content/en/docs/concepts/services-networking/_index.md

```diff
@@ -7,44 +7,214 @@ description: >
+
+* [Supported container runtimes](https://github.com/containernetworking/cni#container-runtimes)
+
+* [Reference plugins](https://www.cni.dev/plugins/current/#reference-plugins)
+
+* [3rd party plugins](https://github.com/containernetworking/cni#3rd-party-plugins)
+
+* [Introduction to CNI](https://youtu.be/YjjrQiJOyME)
+
+* [CNI deep dive](https://youtu.be/zChkx-AB5Xc)
+
+* [Webinar: Kubernetes and Networks](https://www.youtube.com/watch?v=GgCA2USI5iQ)
+
+The examples in the following section illustrate the use of CNI plugins in achieving container network connectivity.
+
+#### Pods on different nodes using a virtual overlay network
+
+This example uses a virtual overlay network to support inter-pod networking.  
+
+{{< figure src="/docs/images/k8net-net-overlay.drawio.svg" alt="k8s pods virtual overlay" class="diagram-large" caption="Figure 4. Example of Pod 1 - Pod 2 networking on different nodes using a virtual overlay network" >}}
+
+In this example, the network plugin has set up an _virtual overlay network_. Packets that need to go
+to a different node are encapsulated and sent over a tunnel mechanism such as
+[GENEVE](https://en.wikipedia.org/wiki/Generic_Network_Virtualization_Encapsulation). 
+
+The destination node retrieves the inner packet and sends it to the target Pod, providing that there are no
+[NetworkPolicy](/docs/concepts/services-networking/network-policies/) restrictions to block that packet.
+
+The virtual overlay approach in this example offers several advantages: 
+
+* Isolates the pod network from the physical underlay network. 
+* Enhanced security with a CNI plugin that handles encrypted communications such as IPsec.
+ 
+A disadvantage to this approach is CNI plugin management, tunnel encap/decap performance overhead and network troubleshooting.
+
+For more information on virtual overlay network implementations, see [Flannel](https://github.com/flannel-io/flannel) and [Weavenet](https://www.weave.works/docs/net/latest/overview/). 
+
+
+#### Pods on different nodes using a physical underlay network
+
+This example uses the _physical underlay network_ to support inter-pod networking. 
+
+{{< figure src="/docs/images/k8net-net-phys-underlay.drawio.svg" alt="k8s pods physical underay" class="diagram-large" caption="Figure 5. Example of Pod 1 - Pod2 networking on different nodes using a physical underlay network" >}}
+
+There several advantages to the physical underlay approach in this example. 
+
+* No need for you to deploy and manage a separate virtual overlay network.   
+* Leverage existing deployment routing protocols such as BGP, address management and operational troubleshooting.  
+
+A disadvantage you might consider is the additional work to protect the pod network.
+
+For more information on physical network underlay implementations, see [Calico BGP](https://github.com/projectcalico/calico) and [Cilium native-routing](https://docs.cilium.io/en/v1.13/network/concepts/routing/#native-routing). 
```

Comment:
:warning: This part doesn't look right. The underlay is jargon to describe the network fabric that the overlay network runs on top of.

For example: _imagine that you get 7 Raspberry Pi computers, plug 'em together with Ethernet cables and switches. That's going to be your underlay network. You can touch it and unplug it, if you want to._

_Now set up [Tailscale](https://tailscale.com/), or something like it. On each Pi._

_Get 7 more Raspberry Pis and connect them to a different subnet, again using Ethernet cables etc. Put them on the same Tailscale-managed overlay network. The overlay lets all the nodes talk to each other, even if the two sets of computers are in different rooms / buildings / countries._

Anyway, if you don't set up an overlay VPN, you've not got an underlay. What you have in the no-VPN, all-in-one-apartment case is “a network”.

The heading to use here could be “**#### Pods on different nodes without using an overlay network**”

Before we get to that heading, though, there is a use for the underlay. You need to talk to the Tailscale server. And it's how you set up basic services like NTP for time, or maybe SSH for diagnostics. We might want to hint at that detail before moving on.

In the example with a GENEVE overlay, the underlay network provides:
- transit for the GENEVE-encapsulated overlay traffic
- communication between nodes and control plane
- NTP (we might as well mention it, accurate time sync is a useful thing to have)
  some VM providers let you sync time from the host machine, mind
- IPsec management (IKEv2 and so forth)

and probably also provides:
- a way for load balancers to reach NodePorts on the VMs

I wouldn't mention BGP; it's a complexity best avoided here.

---

Maybe this scenario is best parked for now. How about explaining the scenario where the kubelet makes an HTTP request to a Pod to check whether it is alive? That's a host-to-pod scenario.

There's also a node-to-control-plane scenario you might want to draw; this'll use the underlay network only.

Reply:
The "underlay" jargon is defined/mentioned in multiple documents such as [Geneve: Generic Network Virtualization Encapsulation RFC 8926](https://datatracker.ietf.org/doc/rfc8926/), [An Architecture for Data-Center Network Virtualization over Layer 3 (NVO3)](https://datatracker.ietf.org/doc/html/rfc8014) and many more. 

IMHO, the overlay/underlay paradigm is a simple pattern that new readers can make sense of, and one that is used in K8s clusters today. I think new diagrams might help visualize this better.

Reply:
It's fine to have an underlay, but it only makes sense when we're also talking about an overlay. If that was the intent, I think the underlay diagram is fine to keep, but with changes - I think from a technical perspective it's not quite telling the right story.

Reply:
Sounds good. Let me hack some changes to better explain the underlay question and a few others you pointed out.